File: 19/Solution.java
class Solution {
    public class ListNode {
        int val;
        ListNode next;
        ListNode() {}
        ListNode(int val) { this.val = val; }
        ListNode(int val, ListNode next) { this.val = val; this.next = next; }
    }
    
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode p1 = head;
        for (int i = 0;i < n; i++) {
            p1 = p1.next;
        }
        if (p1 == null) {
            return head.next;
        }
        ListNode p2 = head;
        while (p1.next != null) {
            p1 = p1.next;
            p2 = p2.next;
        }
        ListNode delNode = p2.next;
        p2.next = delNode.next;
        return head;
    }
}

File: 19/solution.go
package _19

type ListNode struct {
	Val  int
	Next *ListNode
}

func removeNthFromEnd(head *ListNode, n int) *ListNode {
	p1 := head
	for i := 0; i < n; i++ {
		p1 = p1.Next
	}
	if p1 == nil {
		return head.Next
	}
	p2 := head
	for p1.Next != nil {
		p1 = p1.Next
		p2 = p2.Next
	}
	delNode := p2.Next
	p2.Next = delNode.Next
	return head
}
